cloud-governance-mcp/app.py

```
@@ -0,0 +1,770 @@
+                st.code(json.dumps(parsed, indent=2), language="json")
+            except:
+                # If not JSON, show as text (truncate if too long)
+                if len(result) > 2000:
+                    st.text(result[:2000] + f"\n\n... (truncated, total {len(result)} chars)")
+                else:
+                    st.text(result)
+
+    return result
+
+
+def run_agent_loop_gemini(user_message: str, tools: List[types.FunctionDeclaration], previous_messages: list = None) -> str:
```

Ruff (0.15.12)

[warning] PEP 484 prohibits implicit `Optional`

Convert to `T | None`

(RUF013)